### PR TITLE
Fix FooterButton clicking area

### PR DESCRIPTION
**Pull request by Joehuu**
Fixes this. Yellow was on the edge on the old area. Blue was out of the highlighted area. Also pertains between the buttons (between mod and random).

 

---

Should only receive input on the highlighted area.

### CODE REVIEW

File: osu.Game/Screens/Select/FooterButton.cs [hunk 1]
```diff
@@ -55,6 +55,8 @@ public Color4 SelectedColour
         private readonly Box box;
         private readonly Box light;
 
+        public override bool ReceiveMouseInputAt(Vector2 screenSpacePos) => box.ReceiveMouseInputAt(screenSpacePos);
```
Comment (smoogipoo) [inline osu.Game/Screens/Select/FooterButton.cs]: Instead of this, how about moving the shear to the footer button itself?

**Review by smoogipoo — CHANGES_REQUESTED** [review]

File: osu.Game/Screens/Select/FooterButton.cs (hunk 1 above)
Comment (Joehuu) [reply osu.Game/Screens/Select/FooterButton.cs]: If you mean this, then it'll result with:



```
        public FooterButton()
        {
            Shear = shearing;

            Children = new Drawable[]
            {
                box = new Box
                {
                    RelativeSizeAxes = Axes.Both,
                    EdgeSmoothness = new Vector2(2, 0),
                    Colour = Color4.White,
                    Alpha = 0,
                },
                light = new Box
                {
                    Height = 4,
                    EdgeSmoothness = new Vector2(2, 0),
                    RelativeSizeAxes = Axes.X,
                },
                spriteText = new OsuSpriteText
                {
                    Anchor = Anchor.Centre,
                    Origin = Anchor.Centre,
                }
            };
        }
```

**Review by peppy — APPROVED** [review]

**Comment by peppy** [comment]
I'm fine with this approach. @smoogipoo can you check the feedback and follow up?

**Comment by smoogipoo** [comment]
Hmm okay I see.

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**